### PR TITLE
Fix sidebar bug - Sidebar menu not highlighted when active

### LINKED ISSUE #86: [BUG]  sidebar menu is not highlighted when clicked.
### Has this bug been raised before?

- [X] I have checked "open" AND "closed" issues and this is not a duplicate

### Where did you find this bug?

production and dev environment 

### Description

Sidebar menu links are not highlighting upon clicking

### Steps to Reproduce

1. click the menu in the sidebar and it doesn't highlighted the clicked menu.

### Screenshots

_No response_

### Do you want to work on this issue?

Yes

### If "yes" to above, please explain how you would technically implement this

_No response_

### PR DESCRIPTION
**Pull request by myj009**
## Fixes Issue 86
Closes #86 

## Changes proposed

The `current` field in the `navigationMenu` object was assigned a static value `false`.
Have made it a dynamic value based on the current pathname.
Added useMemo to avoid unnecessary recreation of the `navigationMenu` object

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

Updated sidebar

## Summary by CodeRabbit

- **New Features**
	- Side navigation now dynamically highlights the current page.
- **Performance Improvements**
	- Optimized side navigation performance with memoization techniques.

### CODE REVIEW

File: src/components/SideNav.js [hunk 1]
```diff
@@ -77,37 +87,40 @@ export default function SideNav({ setKeyword, children }) {
...
+        id: 4,
+        name: "Logout",
+        href: "/",
+        onClick: logout,
+        initial: "L",
+        current: false,
+      },
+    ],
+    [pathName]
+  );
```
[the remark this answers is not in the document]
Comment (myj009) [reply src/components/SideNav.js:123]: Commented-out code is part of the original code. Did not touch it.

**Comment by myj009** [comment]
Not sure if useMemo will do any real benefit as we are not doing any heavy calculations here
Thoughts??

**Comment by eddiejaoude** [comment]
> Not sure if useMemo will do any real benefit as we are not doing any heavy calculations here
> Thoughts??

yeah probably not in this case. If `useMemo` is not used, can the data be kept in the same location as before?

File: src/components/SideNav.js [hunk 2]
```diff
@@ -18,34 +18,44 @@ import Logo from "@/assets/repo-rater-logo.svg";
...
+        name: "GitHub Repos",
+        href: "/",
+        icon: FolderIcon,
+        current: pathName === "/",
+      },
+      {
+        name: "Popular Repos",
+        href: "/popular",
+        icon: ServerIcon,
+        current: pathName === "/popular",
```
Comment (eddiejaoude) [inline src/components/SideNav.js:40]: This seems like a duplicate 🤔 Can the data be kept outside the function as before and have the calculation in the function?

Comment (myj009) [reply src/components/SideNav.js:40]: Yup. I think we can keep the data outside and do data.map inside the function to fill the current key in the object based on the pathname

Comment (myj009) [reply src/components/SideNav.js]: Done

File: src/components/SideNav.js [hunk 3]
```diff
@@ -83,21 +97,21 @@ export default function SideNav({ setKeyword, children }) {
       name: "Add Rating",
       href: "/rate",
       initial: "+",
-      current: false,
+      current: pathName === "/rate",
```
Comment (eddiejaoude) [inline src/components/SideNav.js:100]: Looking good, are these still needed, because the check is done on line `60` with `href`?

Comment (myj009) [reply src/components/SideNav.js:100]: I can see that we have 2 navigation menus. One is for the public routes and the other is for secure routes.
Have made changes to both
The secure routes were declared inside the function. So have not moved it out of the function.

Comment (eddiejaoude) [reply src/components/SideNav.js:100]: Oh yes you are correct, these should be both done the same way - that was my mistake

I think maybe both can be moved inside the function and have then then the `map` can be removed?

Sorry for the confusion and extra changes

Comment (myj009) [reply src/components/SideNav.js:100]: haha. no worries. I think then my first commit is the right way to go about this issue

Comment (myj009) [reply src/components/SideNav.js:93]: Done

Comment (eddiejaoude) [reply src/components/SideNav.js:93]: thank you 👍

**Review by eddiejaoude — APPROVED** [review]
Looks good 👍  sorry for the confusion

**Pull request merged**